src/Google/Ads/GoogleAds/Lib/V0/LogMessageFormatter.php

```diff
@@ -114,6 +116,14 @@ public function formatDetail(
         );
 
         if ($status->code === 0) {
+            // Initialize unused GoogleAdsFailure object when there are partial failures returned.
+            // This is necessary, as the pool needs to be aware of this class, in order to serialize
+            // the response correctly.
+            if (method_exists($response, 'getPartialFailureError')
+                && !is_null($response->getPartialFailureError())
+                && !isset(self::$unusedGoogleAdsFailure)) {
```

Comment:
Nit: maybe check this first, because after it is initialized the first time it will always short circuit the other checks?

Reply:
+1

Reply:
Done.